src/Data/Lens/At.purs

```diff
@@ -15,6 +16,25 @@ import Data.Newtype (unwrap)
 import Data.Lens (Lens', lens)
 import Data.Lens.Index (class Index)
 
+
+-- | `At` is useful for types like `Data.Map`:
+-- | 
+-- | * The result of getting an element is a `Maybe a`. 
+-- | * New elements can be added.
+-- | * Existing elements can be deleted.
+-- | 
+-- | ```purescript 
+-- | > optic = at "key"
+-- | > view optic $ Map.singleton "key" "value"
+-- | (Just "value")
+-- | 
+-- | > set optic (Just "NEW") $ Map.singleton "key" "value"
+-- | (fromFoldable [(Tuple "key" "NEW")])
+-- | 
+-- | > set optic Nothing $ Map.singleton "key" "value"
+-- | (fromFoldable [])
+-- | ```
```

Comment:
Generally in the documentation we use an "equational" style for examples, rather than pretending we're in the REPL. I much prefer that personally - for example, I'd express the above block as:

``` purescript
view (at "key") (Map.singleton "key" "value") == Just "value"
set (at "key") (Just "NEW") (Map.singleton "key" "value") == fromFoldable [(Tuple "key" "NEW")]
set optic Nothing (Map.singleton "key" "value") == fromFoldable []
```

My argument for that is the code above is exactly executable as PureScript, doesn't assume someone is using the repl, but still allows copying-and-pasting into the repl. In fact, it does moreso than making the `optic` definition and reusing it, as each line is fully contained. Obviously not all examples can be one-liners, but here seems like a prime candidate.

But again, thoughts welcome.

Reply:
I switched to equational style, and used fewer name bindings. I didn't switch all the way because:

1. It drives me batty when I have to scroll horizontally to see all of an example.
2. Too much repetition makes it harder to focus in on what's different between this example and an adjacent one.

I occasionally use `^^^^^^^^^^^` to draw attention to part of a previous line. I did that in the example for `Lens'`.